### PR TITLE
Birthdays in N days, email validation, contacts search, suggest a command when a typo

### PR DESCRIPTION
**Pull request by gregorytereshko**

### CODE REVIEW

File: next_week_birthdays.py [hunk 1]
```diff
@@ -3,52 +3,45 @@
...
+
+
+def get_birthdays_in_days(users, days_from_now = 0):
+    date = datetime.today().date() + timedelta(days=days_from_now)
+    birthday_day_of_week = date.strftime("%A")
+    birthday_str = date.strftime("%d.%m.%Y")
+
+    birthday_users = list(filter(lambda user: (user['birthday'].date().replace(year=date.year) == date), users))
+    user_names = ', '.join(map(lambda user: user['name'], birthday_users))
+    print(f"{birthday_day_of_week} ({birthday_str}): {user_names}")
```
Comment (st3n) [inline next_week_birthdays.py:33]: the same, you should do return here, not print

File: contact_data.py [hunk 2]
```diff
@@ -9,7 +9,14 @@ def __init__(self, value):
 
     def __str__(self):
         return str(self.value)
-
+    
+    def is_eq(self, another_value):
```
Comment (st3n) [inline contact_data.py:13]: what this function should do? please leave a comment
bool(set(arr_value).intersection(splitted_another_value))
here you get rid of repeated values in a list are_value and do intersection with splitted_another_value(without putting it a set) .

**Review by st3n — CHANGES_REQUESTED** [review]
1. please address comments
2. please move all logic not related to validator in a corresponding branches and make separate PR

File: contact_data.py (hunk 2 above)
Comment (gregorytereshko) [reply contact_data.py:13]: @st3n 
Yeah, I agree, I guess the name is a bit off. It must be something like "value_includes()"
It compares a value of a field with another value and returns true if they are equal
It also works for arrays, in this case, it checks for intersections

File: next_week_birthdays.py [hunk 3]
```diff
@@ -3,52 +3,45 @@
 
 
 def get_birthdays_per_week(users):
```
[the remark this answers is not in the document]
Comment (gregorytereshko) [reply next_week_birthdays.py:5]: The logic is different
get_birthdays_per_week - shows, well... next week's birthdays considering all that weekend stuff
get_birthdays_in_days - simply shows birthdays in N days, super simple

File: next_week_birthdays.py [hunk 4]
```diff
@@ -3,52 +3,45 @@
...
+      if next_saturday <= birthday <= next_friday:
+        birthday_day_of_week = birthday.strftime("%A")
+        birthday_str = birthday.strftime("%d.%m.%Y")
+        if birthday.weekday() > 4: # Saturday & Sunday check
+          birthdays[f"Monday ({birthday_str})"].append(user["name"])
+        else:
+          birthdays[f"{birthday_day_of_week} ({birthday_str})"].append(user["name"])
+
+    for day, names in birthdays.items():
+      print(f"{day}: {', '.join(names)}")
```
[the remark this answers is not in the document]
Comment (gregorytereshko) [reply next_week_birthdays.py:23]: Fair enough

File: next_week_birthdays.py [hunk 5]
```diff
@@ -3,52 +3,45 @@
...
-        delta_days = (birthday_this_year - today).days
-
-        # evaluation for the next week
-        if delta_days < 7:
-            today_day_of_week = today.strftime("%A")
-            birthday_day_of_week = (today + timedelta(days=delta_days)).strftime("%A")
-
-            # corner case: skip birthdays of collegues with delta == 6 on the next Monday if today is Sunday
-            # (NOTE: this is not mentioned in task description, recomandation)
-            if today_day_of_week == "Sunday" and birthday_day_of_week == "Saturday":
```
[the remark this answers is not in the document]
Comment (gregorytereshko) [reply next_week_birthdays.py:32]: Fair enough

File: next_week_birthdays.py (hunk 1 above)
Comment (gregorytereshko) [reply next_week_birthdays.py:33]: Fair enough

File: next_week_birthdays.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (st3n) [reply next_week_birthdays.py:5]: OK, I would combine it in one. But let's keep it different if you want

File: contact_data.py (hunk 2 above)
Comment (st3n) [reply contact_data.py:13]: But you are making a set only from one argument. This is what I don't understand. You should wrap in a set both arguments

Comment (gregorytereshko) [reply contact_data.py:13]: Yeah, my bad. Fixed in the separate PR

**Pull request closed without merging**